src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIterator.cs

```diff
@@ -1040,5 +1013,15 @@ public unsafe void CopyFromAscii(string data)
             _block = block;
             _index = blockIndex;
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector<byte> GetVector(byte vectorByte)
+        {
+            // Vector<byte> .ctor doesn't become an intrinsic due to detection issue
+            // However this does cause it to become an intrinsic (with additional multiply and reg->reg copy)
+            // https://github.com/dotnet/coreclr/issues/7459#issuecomment-253965670
+            return Vector.AsVectorByte(new Vector<ulong>(vectorByte * 0x0101010101010101ul));
```

Comment:
Did you want the Vector<byte> here with elements broadcasted? Why not Vector<byte>(vectorByte)?